orquesta/conducting.py

```diff
@@ -730,6 +761,50 @@ def update_task_state(self, task_id, route, event):
             state_ctx = {'__state': self.workflow_state.serialize()}
             current_ctx = dict_util.merge_dicts(current_ctx, state_ctx, True)
 
+            # This decides whether this task would be retried or not when 'retry' is specified
+            may_retry = (
+                hasattr(task_spec, 'retry') and
+                hasattr(task_spec.retry, 'when') and
+                task_spec.has_retry()
+            )
+            if may_retry:
```

Comment:
Instead of doing may_retry and then will_retry, can we check all of them in a single if statement?  I prefer this over multiple indented blocks of codes.
`if task_spec.has_retry() and if number of retry has exhausted and if retry when condition is satisfied:`